### ISSUE TITLE
Not able to use alternate forms of MFA besides Okta Verify

### ISSUE THREAD

**Issue by ddubinsky-udemy** [Nike-Inc/gimme-aws-creds#117]
We have several employees who do not want to use their personal mobile devices for work, so we have provided them with yubikey's for 2fa which work as expected with okta, however when we try to register the key or an alternate form of 2 factor besides Okta Verrify App, we get an error:

```~$ gimme-aws-creds --register_device
Using password from keyring for user@company.com
Multi-factor Authentication required.
Pick a factor:
[ 0 ] Unknown MFA type: email
[ 1 ] Unknown MFA type: u2f
[ 2 ] Okta Verify App: SmartPhone_IPhone: Dennis' iphone
[ 3 ] token:software:totp( OKTA ) : user@company.com
[ 4 ] Unknown MFA type: token:hardware
Selection: 1
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/gimme-aws-creds", line 17, in <module>
    GimmeAWSCreds().run()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gimme_aws_creds/main.py", line 385, in run
    auth_result = okta.auth_session()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gimme_aws_creds/okta.py", line 134, in auth_session
    login_response = self.auth()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/gimme_aws_creds/okta.py", line 126, in auth
    while flow_state.get('apiResponse').get('status') != 'SUCCESS':
AttributeError: 'NoneType' object has no attribute 'get'
~$
```

The same error occurs when I test with 
```
Pick a factor:
[ 0 ] Unknown MFA type: email
[ 1 ] Unknown MFA type: u2f
[ 3 ] token:software:totp( OKTA ) : user@company.com
[ 4 ] Unknown MFA type: token:hardware
```
```
[ 2 ] Okta Verify App: SmartPhone_IPhone: Dennis' iphone 
```
works as expected.  Some help would be greatly appreciated.

Thanks!

**Comment by reversefold** [comment]
It would also be nice to support Duo as an MFA option. As a reference, this tool supports DUO and works properly with Okta and Duo, but is written in Go. https://github.com/Versent/saml2aws

**Comment by mdu42** [comment]
If you're interested, I've implemented for my specific needs U2F  support in my branch https://github.com/mdu42/gimme-aws-creds/tree/WebAuthN

It works either with FIDO1 enabled okta orgs or WebAuthN enabled orgs.
Tested successfully on MacOS and Windows 10

As it uses yubico low-level lib (fido2), it will only works with USB security keys such as yubikeys or any FIDO2 compatible USB security keys. 
For WebAuthN configuration, for moment, it won't work with Authenticators like TouchID or Windows Hello until someone implement them ...
Support for DUO is another bunch of work. I ll look at it if I have time.

Note that this branch probably brakes pipe to shell.

I'm interested in your feedbacks if you test this out :)
And hopefully if everything is stable, planning to make a pull request.
